code/common/src/main/scala/com/avast/cloud/datadog4s/common/Repeated.scala

```diff
@@ -0,0 +1,34 @@
+package com.avast.cloud.datadog4s.common
+
+import java.time.Duration
+
+import cats.effect.{ Concurrent, Resource, Timer }
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.apply._
+import cats.syntax.applicative._
+
+import scala.concurrent.duration._
+
+object Repeated {
+  def run[F[_]: Concurrent: Timer](
+    initDelay: Duration,
+    delay: Duration,
+    iterationTimeout: Duration,
```

Comment:
Not sure if we want to limit users with this. I would leave it up to them.

Reply:
You could say the same thing when it comes to errors. I would like to avoid the scenario where metrics are collected every 10 seconds for example but the operation can last more than than, so i would like to force the user to make a decision. At least in this library i want to make it somewhat opinionated.